lib/streamlit/script_runner.py

```diff
@@ -349,6 +349,19 @@ def _run_script(self, rerun_data):
         finally:
             self._on_script_finished(ctx)
 
+            # Remove references to code variables. Helps to free memory.
+            # We should remove module only after _on_script_finished because
+            # it is used in SCRIPT_STOPPED_WITH_SUCCESS event listeners
+            sys.modules.pop("__main__", None)
+            del module
+
+            # Force garbage collection to run, to help avoid memory use building up
+            # This is usually not an issue, but sometimes GC takes time to kick in and
+            # causes apps to go over resource limits, and forcing it to run between
+            # script runs is low cost, since we aren't doing much work anyway.
+            if config.get_option("runner.postScriptGC"):
+                gc.collect(2)
```

Comment:
Curious on why you chose to move this code into the `finally` clause, rather than just use at the end of `_in_script_finished`. It seems like it would benefit being in the function?

Alternatively, we can create a clean_up function specific for this, but may be overkill.

Reply:
@kmcgrady it is important to call `sys.modules.pop("__main__", None); del module` after `_on_script_finished` as described in comment. So I've moved this code part outside this method, just to be sure that any new code inside `_on_script_finished` method will be added before memory cleanup. It will make others developers think about why this code is outside `_on_script_finished` and avoid mistakes. 

I can move it into a separate method as you have proposed.